src/main/java/com/squareup/javapoet/AnnotationSpec.java

```diff
@@ -221,7 +222,9 @@ public Builder addMember(String name, CodeBlock codeBlock) {
      * depending on the given {@code value} object. Falls back to {@code "$L"} literal format if
      * the class of the given {@code value} object is not supported.
      */
-    private Builder addValue(String memberName, Object value) {
+    public Builder addMemberForValue(String memberName, Object value) {
```

Comment:
Hmmm . . . . is our approach wrong here?

With this API we offer two ways to do everything:
```
  builder.addMember("foo", "$S", "hello");
  builder.addMemberForValue("foo", "hello");
  builder.addMember("bar", "$T.$N", Roshambo.class, Roshambo.ROCK);
  builder.addMemberForValue("bar", Roshambo.ROCK);
```
Is it better to permit a single API?

```
  builder.addMember("foo", "$S", "hello");
  builder.addMember("foo", "$V", "hello");
  builder.addMember("bar", "$T.$N", Roshambo.class, Roshambo.ROCK);
  builder.addMember("bar", "$V", Roshambo.ROCK);
```